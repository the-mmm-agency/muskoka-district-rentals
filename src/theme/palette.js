const palette = {
  primary: '#212a38',
  secondary: '#3278b5',
  alt: '#718198',
  background: {
    dark: '#eaecef',
    default: '#f1f1f1',
    light: '#ffffff',
  },
  text: {
    primary: '#344051',
    secondary: '#999999',
  },
}

export default palette
