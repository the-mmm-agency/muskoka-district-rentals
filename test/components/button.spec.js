import React from 'react'
import renderer from 'react-test-renderer'
import { matchers } from 'jest-emotion'

import Button from 'components/button'
import palette from 'theme/palette'

expect.extend(matchers)

test('it matches the snapshot', () => {
  const tree = renderer.create(<Button>Test Button</Button>).toJSON()
  expect(tree).toMatchSnapshot()
})

test('it applies styles according to passed props', () => {
  const tree = renderer
    .create(
      <Button color="secondary" font="serif" radius="curved" size="small">
        Test Button
      </Button>
    )
    .toJSON()
  expect(tree).toHaveStyleRule('background', palette.secondary)
  expect(tree).toHaveStyleRule('font-family', 'Playfair Display')
  expect(tree).toHaveStyleRule('font-size', '0.65rem')
  expect(tree).toHaveStyleRule('border-radius', '6px')
})
