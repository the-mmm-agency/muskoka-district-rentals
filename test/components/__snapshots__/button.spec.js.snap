// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it matches the snapshot 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border-radius: 0px;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-weight: 500;
  outline: none;
  margin: 0;
  padding: 0;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  vertical-align: middle;
  -webkit-transition: border 250ms cubic-bezier(0.4,0,0.2,1) 0ms,background 250ms cubic-bezier(0.4,0,0.2,1) 0ms,box-shadow 250ms cubic-bezier(0.4,0,0.2,1) 0ms,-webkit-transform 250ms cubic-bezier(0.4,0,0.2,1) 0ms;
  -webkit-transition: border 250ms cubic-bezier(0.4,0,0.2,1) 0ms,background 250ms cubic-bezier(0.4,0,0.2,1) 0ms,box-shadow 250ms cubic-bezier(0.4,0,0.2,1) 0ms,transform 250ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: border 250ms cubic-bezier(0.4,0,0.2,1) 0ms,background 250ms cubic-bezier(0.4,0,0.2,1) 0ms,box-shadow 250ms cubic-bezier(0.4,0,0.2,1) 0ms,transform 250ms cubic-bezier(0.4,0,0.2,1) 0ms;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  background: #212a38;
  border: none;
  color: white;
  font-size: 0.875rem;
  padding: 10px 16px;
}

.c0::-moz-focus-inner {
  border-style: 'none';
}

.c0:hover {
  background: #212a38;
  box-shadow: 0px 1px 5px 0px rgba(0,0,0,0.2),0px 2px 2px 0px rgba(0,0,0,0.14),0px 3px 1px -2px rgba(0,0,0,0.12);
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transform: translateY(-3px);
  -ms-transform: translateY(-3px);
  transform: translateY(-3px);
}

@media (hover:none) {
  .c0:hover {
    background-color: transparent;
  }
}

<button
  className="c0"
  color="primary"
  radius="default"
  size="medium"
>
  Test Button
</button>
`;
